### ISSUE TITLE
Support For Gradient Boosting Methods

### ISSUE THREAD

**Issue by vyathakavilocana** [Palashio/libra#187]
Much of machine learning work with tabular and structured data in a professional or competitive setting employs the usage of gradient boosting methods for classification/regression purposes. We should consider adding support for certain methods such as XGBoost, CatBoost, and LightGBM given these appear to the most cross-functional and popular gradient boosting libraries on offer.
